src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/McpServer/McpServer-CSharp/Tools/RandomNumberTools.cs

```diff
@@ -0,0 +1,49 @@
+using System.ComponentModel;
+using ModelContextProtocol.Server;
+
+namespace McpServer_CSharp.Tools;
+
+/// <summary>
+/// Sample MCP tools for demonstration purposes.
+/// These tools can be invoked by MCP clients to perform various operations.
+/// </summary>
+public class RandomNumberTools
+{
+    private readonly int _maxNumber;
+
+    public RandomNumberTools()
+    {
+        // Process configuration settings from the environment variables.
+        // These will be provided by the MCP client application, such as VS Code.
+        // Configuration settings could be provided via dependency injection and the IOptions pattern.
+        var maxNumberEnv = Environment.GetEnvironmentVariable("MAX_RANDOM_NUMBER");
+        if (!int.TryParse(maxNumberEnv, out var maxNumber) || maxNumber <= 0)
+        {
+            throw new InvalidOperationException("Error: you must set the MAX_RANDOM_NUMBER environment variable to a positive integer.");
+        }
+
+        _maxNumber = maxNumber;
+    }
```

Comment:
Could we simplify this further, by deleting all this goop, and instead just have the tool accept a max value as a parameter? That'd be less code, more configurable from a client use perspective, and demonstrate how you can use parameters in tools, with schemas automatically generated, parsing automatically performed, etc., e.g.
```C#
[McpServerTool]
[Description("Returns a random number between 1 and the maximum number allowed by the tool.")]
public int GetRandomNumber(
    [Description("Inclusive maximum number")] int maxNumber = int.MaxValue - 1)
{
    return Random.Shared.Next(1, maxNumber + 1);
}
```

Reply:
Thoughts on this @timheuer and @MackinnonBuck? Do we want to demonstrate the configuration experience (environment variable) or not in the template?

Reply:
I can see both value.  The env hits the mark on how people provide API keys, etc. but this use case isn't really that...so perhaps better for a doc area/tutorial later.  ALl in favor of simplifying.  The bones of the structure is what people want out of a template.  v1 -- let's keep it even simpler then.  Evolve and perhaps could put a doc link in the template to more concepts (i.e., for more scenarios https://aka.ms/dotnet-mcp-server-tutorial)

Reply:
Okay, sounds good. I'll remove the goop!

Reply:
Done.